### PR TITLE
Resource counts

### PR DESCRIPTION
**Pull request by SenecaAnderson**
### Summary
When you click on the "Patient x" dropdowns in the sidebar of the output display, the accordion expands and displays a list of resources and the number of each. The total number of resources is at the bottom of the list.

### Code Changes
Several functions were added to the Result component, and the JSX returned by it was modified. A CSS class was added to Pages.scss to make the "Total Resources: x" have more visible styling.

### New Behavior
When you click on the "Patient x" dropdowns, a list of resources and the number of each appears.

### Testing Guidance
Use npm start to start the app. Click on "Extract New". Provide a valid config file (as well as anything else you'd like), then click "Submit". Click on the "Patient x" dropdowns in the sidebar to toggle between patients and view the resource counts.

### CODE REVIEW

File: react-app/src/components/Result.js
Comment (Dtphelan1) [inline react-app/src/components/Result.js:11]: DP1 - We could probably drop this right?

File: react-app/src/components/Result.js
Comment (Dtphelan1) [inline react-app/src/components/Result.js:16]: DP2 - Does this need to live above the forEach loop? Could we just create this string within the body of the loop?  It's was a little confusing that we have a function that returns a string instead of just having a string that uses string interpolation in the forEach loop

File: react-app/src/components/Result.js
Comment (Dtphelan1) [inline react-app/src/components/Result.js:29]: DP3 - It's discouraged to use index-in-an-array as a key for a `.map`. The reasons for it being less-than-ideal won't materialize here, but we should really be using something that is unique to the thing being rendered in the iteration of the loop (for example, the resourceType!). If you want to read more look here: https://robinpokorny.medium.com/index-as-a-key-is-an-anti-pattern-e0349aece318

File: react-app/src/components/Result.js
Comment (Dtphelan1) [inline react-app/src/components/Result.js:55]: DP4 - I think I'd prefer Total Resources above the count-by-each-resource-type

**Review by Dtphelan1 — CHANGES_REQUESTED** [review]
Let me know when those changes are done and I'll take another look!

File: react-app/src/components/Result.js
Comment (SenecaAnderson) [reply react-app/src/components/Result.js:16]: This is function is a modified version of the getResourceCountInBundle function from /mcode-extraction-framework/src/helpers/fhirUtils.js. I deleted it from my code and it works fine. If you want I could make the same change in the original function and submit a pull request.

Comment (Dtphelan1) [reply react-app/src/components/Result.js]: I don't think you need to worry about making the change over there, making the change here is sufficient for this task.

**Review by Dtphelan1 — APPROVED** [review]
This looks good to me count-wise. Even after we get a second approval here, we will want to hold-off merge until the other PR is merged in, and then change the base branch here to be `master`.

**Review by jafeltra — COMMENTED** [review]
I just had a few comments, but this looks good!

File: react-app/src/components/Result.js
Comment (jafeltra) [inline react-app/src/components/Result.js:49]: I have one more question for you and @Dtphelan1. I think previously we have been trying to keep from assuming that the id we display will be an integer, so I'm not sure if we should be incrementing this value here. Was there a reason you decided to change it here? It seems like it might be slightly more complicated to change the id we pass in based on how it gets used in other places, so before we decide to change things, I'm curious if there was a reason for the change and what you and Dylan both think about displaying a `Patient 0` vs `Patient 1` for the first patient.

Comment (SenecaAnderson) [reply react-app/src/components/Result.js:49]: The id that's displayed is the iterator variable from here, so it's guaranteed to be an int:
`const list = props.extractedData.map((bundle, i) => (
    <Result bundle={bundle} id={i} setPatientID={props.setPatientID} key={i} setShowLogs={props.setShowLogs} />
  ));`

In ResultPage, patientID (which is modified whenever someone clicks on a patient) is used as an index variable to determine which JSON schema gets passed to PatientData:
`<PatientData
              patientJson={props.extractedData[patientID]}
              id={patientID}
              showLogs={showLogs}
              loggedMessages={props.loggedMessages}
            /><PatientData
              patientJson={props.extractedData[patientID]}
              id={patientID}
              showLogs={showLogs}
              loggedMessages={props.loggedMessages}
            />`

Since it's generated by calling extractedData.map() used to access bundles in the extractedData array, the count starts at 0. However, I thought starting with 0 in the output might seem strange/confusing to people who don't have a computer science background.

Comment (SenecaAnderson) [reply react-app/src/components/Result.js:49]: The 'Patient X' thing was also originally intended to be a placeholder for some more useful title. If you want, I could use fhirpath to retrieve the patient's name or mrn id, and display that instead.

Comment (jafeltra) [reply react-app/src/components/Result.js:49]: I definitely agree with you on both points - right now the id is guaranteed to be an integer and it would probably be nice to not index a list by zero. However, I think [these comments on previous PRs](https://github.com/mcode/mcode-extraction-ui/pull/3#discussion_r653012115) also discuss how even though we are only using an integer for right now, we want to make sure we aren't locked into using an integer id. This would be another example of when we want to be a little more flexible around how we develop this component to help us ensure we can support different types of data in the future.

I do really like the idea of using `fhirpath` to retrieve the name or another piece of information here. We can discuss at stand up (which is right now haha) about if we want to do that now or if we want to revert this line and do the fhirpath piece separately.

Comment (jafeltra) [reply react-app/src/components/Result.js]: Following up on the conversation we had at stand up, we'll plan to change back the display here to just be `props.id` so we avoid requiring an integer, but I'll make a task to explore what might the best way to label each patient in the side bar.

**Review by jafeltra — APPROVED** [review]
This looks good to me! We'll just have to rebase this once the date picker branch is merged.

**Comment by jafeltra** [comment]
I know I said we will need to rebase, but after talking to @Dtphelan1 I think the rebase might not work well since we have so many branches off of each other. We probably should be using a merge instead of a rebase, but we can discuss as a group when it's time to update.

**Pull request merged**
